pandas/tests/test_frame.py

```diff
@@ -3568,6 +3565,20 @@ def test_constructor_tuples(self):
         expected = DataFrame({'A': Series([(1, 2), (3, 4)])})
         assert_frame_equal(result, expected)
 
+    def test_constructor_namedtuples(self):
+        # GH11181
+        from collections import namedtuple
+        named_tuple = namedtuple("Pandas", list('ab'))
+        tuples = [named_tuple(1, 3), named_tuple(2, 4)]
+        expected = DataFrame({'a': [1, 2], 'b': [3, 4]})
+        result = DataFrame(tuples)
+        assert_frame_equal(result, expected)
+
```

Comment:
That's pretty expensive:
```python
In [1]:

from collections import namedtuple
nt = namedtuple('NT', list('abc'))
tuples = [nt(0,1,2) for i in range(int(1e7))]
In [2]:

t=tuples[0]
correct_type=type(t)
In [3]:

%timeit all(type(tup)==correct_type for tup in tuples)
1 loops, best of 3: 1.16 s per loop
```
Given this is a 'best efforts' check - i.e. the alternative is 'useless' columns of `(0, 1, 2)` - is taking the `_fields` from the first item OK? If the user really cares, she can supply `columns`, if not, she gets something reasonable...